### PR TITLE
Preview/1926 responsibility separation

### PR DESCRIPTION
**Pull request by tnickelsen**

### CODE REVIEW

File: domains/measurements/Measurements.API/API/Extensions/IConfigurationExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Serilog;
+using Serilog.Enrichers.Span;
+using Serilog.Formatting.Json;
+
+namespace API.Extensions;
+
+public static class IConfigurationExtensions
```
Comment (ckr123) [inline domains/measurements/Measurements.API/API/Extensions/IConfigurationExtensions.cs:10]: Why this new startup pattern? All our other sub systems are not using it and now we have 3 classes just to start the app. Whats wrong with program.cs? :)

File: domains/measurements/Measurements.API/API/Measurements/Helpers/TimeSeriesHelper.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    {
+        private const string TimeZoneId = "Romance Standard Time";
+
+        public static DateTimeOffset ConvertDanishDateToDateTimeOffset(this string date)
+        {
+            DateTime dt = DateTime.Parse(date);
+            return TimeZoneInfo.ConvertTimeToUtc(DateTime.SpecifyKind(dt, DateTimeKind.Unspecified), TimeZoneInfo.FindSystemTimeZoneById(TimeZoneId));
+        }
+
+        public static DateTimeOffset ConvertToDanishTimezone(this DateTimeOffset date)
```
Comment (ckr123) [inline domains/measurements/Measurements.API/API/Measurements/Helpers/TimeSeriesHelper.cs:17]: This is not beeing used

File: domains/measurements/Measurements.API/API/Measurements/gRPC/V1/Services/MeasurementsService.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+using Measurements.V1;
+using Metertimeseries.V1;
+
+namespace API.Measurements.gRPC.V1.Services;
+
+public class MeasurementsService : global::Measurements.V1.Measurements.MeasurementsBase
+{
+    private readonly MeterTimeSeries.MeterTimeSeriesClient _client;
+
+    public MeasurementsService(Metertimeseries.V1.MeterTimeSeries.MeterTimeSeriesClient client)
```
Comment (ckr123) [inline domains/measurements/Measurements.API/API/Measurements/gRPC/V1/Services/MeasurementsService.cs:15]:
```suggestion
public MeasurementsService(MeterTimeSeries.MeterTimeSeriesClient client)
```

File: domains/measurements/Measurements.API/API/Measurements/gRPC/V1/Services/MeasurementsService.cs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                        Gsrn = mp.MeteringPointId,
+                        DateFrom = group.Min(m => m.DateFrom),
+                        DateTo = group.Max(m => m.DateTo),
+                        Quantity = group.Sum(m => m.Quantity),
+                        Quality = group.Max(m => m.Quality),
+                    })
+                .Where(measurement => measurement.DateFrom >= request.DateFrom && measurement.DateTo <= request.DateTo)
+                 ?? Enumerable.Empty<Measurement>()
+            ) ?? Enumerable.Empty<Measurement>()
+        ) ?? Enumerable.Empty<Measurement>();
```
Comment (ckr123) [inline domains/measurements/Measurements.API/API/Measurements/gRPC/V1/Services/MeasurementsService.cs:69]: My IDE tells me line`69` is redundant.

File: domains/measurements/Measurements.API/API/Measurements/Helpers/TimeSeriesHelper.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Globalization;
+using Measurements.V1;
+
+namespace API.Measurements.Helpers
+{
+    public static class TimeSeriesHelper
```
Comment (ckr123) [inline domains/measurements/Measurements.API/API/Measurements/Helpers/TimeSeriesHelper.cs:7]: i am guessing all of this is some sort of datahub and how they handle time. Do you know what it all means?

File: domains/measurements/Measurements.API/API/Measurements/Helpers/TimeSeriesHelper.cs [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            }
+            else if (meterReadingOccurrence == "PT15M")
+            {
+                return danishDate.AddMinutes(timePosition * 15).ToUnixTimeSeconds();
+            }
+
+            throw new NotImplementedException($"Meter reading occurance '{meterReadingOccurrence}' is not implemented.");
+        }
+
+        public static int GetQuantityFromMeterReading(string energyTimeSeriesMeasureUnit, string energyQuantity)
```
Comment (ckr123) [inline domains/measurements/Measurements.API/API/Measurements/Helpers/TimeSeriesHelper.cs:45]: I dont think this should be in a TimeSeriesHelper

File: domains/measurements/Measurements.API/API/Measurements/Helpers/TimeSeriesHelper.cs [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+            if (energyTimeSeriesMeasureUnit == "KWH")
+            {
+                return (int)Math.Round(quantityAsDouble * 1000);
+            }
+
+            throw new NotImplementedException($"Measure unit '{energyTimeSeriesMeasureUnit}' not implemented!");
+        }
+
+        public static EnergyQuantityValueQuality GetQuantityQualityFromMeterReading(string QuantityQuality)
```
Comment (ckr123) [inline domains/measurements/Measurements.API/API/Measurements/Helpers/TimeSeriesHelper.cs:57]: I dont think this should be in a TimeSeriesHelper

File: domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Dto/Responses/Enums/MeterType.cs [hunk 8]
```diff
@@ -0,0 +1,12 @@
+using System.Text.Json.Serialization;
+
+namespace API.MeteringPoints.Api.v2024_01_10.Dto.Responses.Enums
+{
+    [JsonConverter(typeof(JsonStringEnumConverter))]
+    public enum MeterType
+    {
+        consumption,
```
Comment (ckr123) [inline domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Dto/Responses/Enums/MeterType.cs:8]: Please capitalize

File: domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Dto/Responses/Address.cs [hunk 9]
```diff
@@ -0,0 +1,21 @@
+namespace API.MeteringPoints.Api.v2024_01_10.Dto.Responses;
+
+public class Address
```
Comment (ckr123) [inline domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Dto/Responses/Address.cs:3]: Why not a record here like the rest?

File: domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Dto/Responses/MeteringPoint.cs [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+        }
+        else if (typeOfMp.StartsWith("D") && typeOfMp.Length == 3 && int.Parse(typeOfMp.Substring(1)) >= 1)
+        {
+            return MeterType.child;
+        }
+
+        throw new NotSupportedException($"TypeOfMP '{typeOfMp}' is not supported.");
+    }
+
+    public static AssetTypeEnum GetAssetType(string assetType)
```
Comment (ckr123) [inline domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Dto/Responses/MeteringPoint.cs:84]: Is Eloverblik using this? If not we should only use the AIB

File: domains/measurements/Measurements.API/API/Shared/Exceptions/SanitizedException.cs [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+namespace API.Shared.Exceptions;
+
+[Serializable]
+public abstract class SanitizedException : Exception
+{
+    private static string? clean(string? message)
+    {
+        if (message == null) return null;
+
+        message = Regex.Replace(message, "([0-9]{10})", "**********");
```
Comment (ckr123) [inline domains/measurements/Measurements.API/API/Shared/Exceptions/SanitizedException.cs:13]: Why this?

File: domains/measurements/Measurements.API/API/Program.cs [hunk 12]
```diff
@@ -1,69 +1,33 @@
...
+using Microsoft.Extensions.Configuration;
 using Serilog;
-using Serilog.Enrichers.Span;
-using Serilog.Formatting.Json;
+using System;
 
-[assembly: InternalsVisibleTo("Tests")]
+var environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
+var configuration = new ConfigurationBuilder()
+    .AddJsonFile("appsettings.json", optional: false)
```
Comment (ckr123) [inline domains/measurements/Measurements.API/API/Program.cs:9]: Why this?

File: domains/measurements/Measurements.API/Tests/Extensions/NSubstituteExtensions.cs [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+using Grpc.Core.Testing;
+using NSubstitute;
+using NSubstitute.Core;
+using System.Threading.Tasks;
+
+namespace Tests.Extensions;
+
+public static class NSubstituteExtensions
+{
+    public static ConfiguredCall Returns<TResponse>(this AsyncUnaryCall<TResponse> value, TResponse response)
```
Comment (ckr123) [inline domains/measurements/Measurements.API/Tests/Extensions/NSubstituteExtensions.cs:11]: Why is this needed?

File: domains/measurements/Measurements.API/Tests/Fixtures/TestServerHelpers/ForwardLoggingProvider.cs [hunk 14]
```diff
@@ -0,0 +1,74 @@
+#region Copyright notice and license
+// Copyright 2019 The gRPC Authors
```
Comment (ckr123) [inline domains/measurements/Measurements.API/Tests/Fixtures/TestServerHelpers/ForwardLoggingProvider.cs:2]: What is this?

File: domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Controllers/MeteringPointsController.cs [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+
+    public MeteringPointsController(Meteringpoint.V1.Meteringpoint.MeteringpointClient client)
+    {
+        _client = client;
+    }
+
+    /// <summary>
+    /// Get metering points from DataHub2.0
+    /// </summary>
+    /// <response code="200">Successful operation</response>
```
Comment (ckr123) [inline domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Controllers/MeteringPointsController.cs:29]: Should it be documented that 200 also could be an empty list if no meteringpooint exists?

File: domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Controllers/MeteringPointsController.cs [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+        var request = new Meteringpoint.V1.OwnedMeteringPointsRequest
+        {
+            Subject = user.Subject.ToString(),
+            Actor = user.Id.ToString()
+        };
+        var response = await _client.GetOwnedMeteringPointsAsync(request);
+
+        var meteringPoints = response.MeteringPoints
+            .Where(mp => MeteringPoint.GetMeterType(mp.TypeOfMp) != MeterType.child)
+            .Select(MeteringPoint.CreateFrom)
```
Comment (ckr123) [inline domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Controllers/MeteringPointsController.cs:46]: Does all parameters get correctly mapped? Looking at the verified test file it seems that something is missing.

File: domains/measurements/Measurements.API/API/Startup.cs [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddHealthChecks();
+
+        services.AddControllers().AddJsonOptions(options =>
+        {
+            options.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter());
+            options.JsonSerializerOptions.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
```
Comment (ckr123) [inline domains/measurements/Measurements.API/API/Startup.cs:39]: Is this the same naming policy throughout all our subsystems?

File: domains/measurements/Measurements.API/API/Startup.cs [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+            var options = sp.GetRequiredService<IOptions<DataHubFacadeOptions>>().Value;
+            o.Address = new Uri(options.Url);
+        });
+        services.AddApiVersioning(options =>
+        {
+            options.AssumeDefaultVersionWhenUnspecified = false;
+            options.ReportApiVersions = true;
+            options.ApiVersionReader = new HeaderApiVersionReader("EO_API_VERSION");
+        })
+            .AddMvc()
```
Comment (ckr123) [inline domains/measurements/Measurements.API/API/Startup.cs:69]: Seems like the indention gone wrong

File: domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs [hunk 19]
```diff
@@ -0,0 +1,229 @@
+#region Copyright notice and license
+// Copyright 2019 The gRPC Authors
```
Comment (ckr123) [inline domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs:2]: Why this?

File: domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs [hunk 20]
```diff
@@ -0,0 +1,229 @@
...
+
+namespace Tests.Fixtures
+{
+    public delegate void LogMessage(LogLevel logLevel, string categoryName, EventId eventId, string message, Exception? exception);
+
+    public class TestServerFixture<TStartup> : IDisposable where TStartup : class
+    {
+        public byte[] PrivateKey { get; set; } = RsaKeyGenerator.GenerateTestKey();
+
+        private TestServer? _server;
```
Comment (ckr123) [inline domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs:44]: it seems that you are using _ on all the private fields. It is not a naming convention we are using in EO. Why differ?

File: domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs [hunk 21]
```diff
@@ -0,0 +1,229 @@
...
+                60,
+                claims
+            );
+
+            return signedJwtToken;
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
```
Comment (ckr123) [inline domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs:199]: Why these if statements? cant we just dispose when the IDisposable.Dispose is hit?

File: domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs [hunk 22]
```diff
@@ -0,0 +1,229 @@
...
+        private HttpMessageHandler? _handler;
+        private GrpcChannel? _channel;
+        private Dictionary<string, string?>? _configurationDictionary;
+        private bool _disposed = false;
+        public event Action<IServiceCollection>? ConfigureTestServices;
+
+        public event LogMessage? LoggedMessage;
+        public GrpcChannel Channel => _channel ??= CreateGrpcChannel();
+
+        public TestServerFixture()
```
Comment (ckr123) [inline domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs:55]: Is this class to setup the Grpc server for test?

File: domains/measurements/Measurements.API/Tests/Fixtures/TestServerHelpers/ForwardLoggingProvider.cs [hunk 23]
```diff
@@ -0,0 +1,74 @@
...
+//
+// https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/grpc/test-services/sample/Tests/Server/IntegrationTests/Helpers/ForwardingLoggerProvider.cs
+#endregion
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Tests.Fixtures.TestServerHelpers
+{
+    internal class ForwardingLoggerProvider : ILoggerProvider
```
Comment (ckr123) [inline domains/measurements/Measurements.API/Tests/Fixtures/TestServerHelpers/ForwardLoggingProvider.cs:24]: Getting a warning regarding IDisposable. I am also unsure what this class actual does

File: domains/measurements/Measurements.API/Tests/Fixtures/TestServerHelpers/TestServerContext.cs [hunk 24]
```diff
@@ -0,0 +1,55 @@
+#region Copyright notice and license
```
Comment (ckr123) [inline domains/measurements/Measurements.API/Tests/Fixtures/TestServerHelpers/TestServerContext.cs:1]: Why this?

File: domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs (hunk 22 above)
Comment (ckr123) [reply domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs:55]: I am quite unsure on the Test.Fixture namespace. I think we should be really specific in the naming on what it does.

File: domains/measurements/Measurements.API/Tests/Measurements/Helpers/TimeSeriesHelperTests.cs [hunk 25]
```diff
@@ -0,0 +1,33 @@
...
+    [InlineData("2022-01-01T12:34:56.789+02:00", "2022-01-01T10:00:00.000+00:00")]
+    public void ZeroedHour_GivenInput_ZeroOffsetToUtc(string inputTimestamp, string expectedTimestamp)
+    {
+        var actual = DateTimeOffset.Parse(inputTimestamp).ToUnixTimeSeconds().ZeroedHour();
+
+        Assert.Equal(DateTimeOffset.Parse(expectedTimestamp), actual);
+        Assert.Equal(TimeSpan.Zero, actual.Offset);
+    }
+
+    [Theory]
```
Comment (ckr123) [inline domains/measurements/Measurements.API/Tests/Measurements/Helpers/TimeSeriesHelperTests.cs:21]: Why make this a theory when only one input is being used?

File: domains/measurements/Measurements.API/Tests/MeasurementsTestBase.cs [hunk 26]
```diff
@@ -0,0 +1,66 @@
...
+            {"TokenValidation:Issuer", "Issuer"},
+            {"TokenValidation:Audience", "Audience"},
+            {"DataHubFacade:Url", DataHubFacadeUrl},
+            {"DataSync:Endpoint", "https://example.com"}
+        };
+
+        _serverFixture.ConfigureHostConfiguration(config);
+    }
+
+    protected virtual void Dispose(bool disposing)
```
Comment (ckr123) [inline domains/measurements/Measurements.API/Tests/MeasurementsTestBase.cs:45]: Why this?

File: domains/measurements/Measurements.API/Tests/MeteringPoints/Api/v2024_01_10/MeteringPointControllerTests.cs [hunk 27]
```diff
@@ -0,0 +1,201 @@
...
+using Tests.Fixtures;
+using System.Linq;
+using System.Net;
+
+namespace Tests.MeteringPoints.Api.v2024_01_10;
+
+[UsesVerify]
+public class MeteringPointControllerTests : MeasurementsTestBase
+{
+    public MeteringPointControllerTests(TestServerFixture<Startup> serverFixture)
```
Comment (ckr123) [inline domains/measurements/Measurements.API/Tests/MeteringPoints/Api/v2024_01_10/MeteringPointControllerTests.cs:22]: instead of a TestServerFixture, cant we use the build-in IClassFixture?

File: domains/measurements/Measurements.API/Tests/MeteringPoints/Api/v2024_01_10/MeteringPointControllerTests.cs [hunk 28]
```diff
@@ -0,0 +1,201 @@
...
+        _serverFixture.ConfigureTestServices += services =>
+        {
+            var mpClient = services.Single(d => d.ServiceType == typeof(Meteringpoint.V1.Meteringpoint.MeteringpointClient));
+            services.Remove(mpClient);
+            services.AddSingleton(clientMock);
+        };
+
+        var client = _serverFixture.CreateAuthenticatedClient(Guid.NewGuid().ToString());
+
+        var response = await client.GetFromJsonAsync<GetMeteringPointsResponse>("api/meteringpoints");
```
Comment (ckr123) [inline domains/measurements/Measurements.API/Tests/MeteringPoints/Api/v2024_01_10/MeteringPointControllerTests.cs:192]: Cant we use verifier?

File: domains/measurements/Measurements.API/Tests/Measurements/gRPC/V1/Services/MeasurementsServiceTests.GetMeasurements.verified.txt [hunk 29]
```diff
@@ -0,0 +1,10 @@
+﻿{
+  Measurements: [
+    {
+      Gsrn: 1234567890123456,
+      DateFrom: 1609542000,
+      DateTo: 1609545600,
+      Quantity: 880
```
Comment (ckr123) [inline domains/measurements/Measurements.API/Tests/Measurements/gRPC/V1/Services/MeasurementsServiceTests.GetMeasurements.verified.txt:7]: Wont we need some more data from measurements when using the gRPC endpint?

File: domains/measurements/Measurements.API/Tests/MeteringPoints/Api/v2024_01_10/MeteringPointControllerTests.GetMeteringPoints.verified.txt [hunk 30]
```diff
@@ -0,0 +1,20 @@
+﻿{
+  Result: [
+    {
+      GSRN: 1234567890123456,
```
Comment (ckr123) [inline domains/measurements/Measurements.API/Tests/MeteringPoints/Api/v2024_01_10/MeteringPointControllerTests.GetMeteringPoints.verified.txt:4]: I think we need to tell if it is production or consumption

File: domains/measurements/Measurements.API/Tests/Models/MeteringPointTests.cs [hunk 31]
```diff
@@ -0,0 +1,60 @@
...
+        [InlineData("D99", MeterType.child)]
+        public void MeteringPoints_MeterType_Success(string stringType, MeterType targetType)
+        {
+            var mt = MeteringPoint.GetMeterType(stringType);
+
+            Assert.Equal(targetType, mt);
+        }
+
+        [Fact]
+        public void MeteringPoints_MeterTypeChildSeries_Success()
```
Comment (ckr123) [inline domains/measurements/Measurements.API/Tests/Models/MeteringPointTests.cs:24]: Can this test be more descriptive in its naming?

File: domains/measurements/Measurements.API/Tests/SwaggerTests.GetSwaggerDocs_v1.0.verified.txt [hunk 32]
```diff
@@ -0,0 +1,211 @@
+﻿{
+  "openapi": "3.0.1",
+  "info": {
+    "title": "Measurements API 1.0",
+    "description": "",
+    "version": "1.0"
+  },
+  "paths": {
+    "/api/measurements/consumption": {
```
Comment (ckr123) [inline domains/measurements/Measurements.API/Tests/SwaggerTests.GetSwaggerDocs_v1.0.verified.txt:9]: would it make sense to make one endpoint for all measurements and put a query param saying if it should be production or consumption?

These endpoints looks a lot like each other and i am guessing if we change something for consumption we should change it on production. So to avoid this i am suggestion making one endpoint

File: domains/measurements/Measurements.API/API/Extensions/IConfigurationExtensions.cs (hunk 1 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/Extensions/IConfigurationExtensions.cs:10]: Because it is needed to use the TestServerFixture. And that is needed to make the gRPC integration tests. The wallet does the same

File: domains/measurements/Measurements.API/API/Measurements/Helpers/TimeSeriesHelper.cs (hunk 2 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/Measurements/Helpers/TimeSeriesHelper.cs:17]: Removed

File: domains/measurements/Measurements.API/API/Measurements/gRPC/V1/Services/MeasurementsService.cs (hunk 3 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/Measurements/gRPC/V1/Services/MeasurementsService.cs:15]: Removed

File: domains/measurements/Measurements.API/API/Measurements/gRPC/V1/Services/MeasurementsService.cs (hunk 4 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/Measurements/gRPC/V1/Services/MeasurementsService.cs:69]: That is under the assumption that you can't set gRPC messages to null. I'm note sure - I've seen it done in an integration test, so I'll leave it just in case

File: domains/measurements/Measurements.API/API/Measurements/Helpers/TimeSeriesHelper.cs (hunk 5 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/Measurements/Helpers/TimeSeriesHelper.cs:7]: It is a helper for MeterTimeSeries yes (which is the response from DataHubFacade). It is taken from DataSync.API. I know what they mean, yes

File: domains/measurements/Measurements.API/API/Measurements/Helpers/TimeSeriesHelper.cs (hunk 6 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/Measurements/Helpers/TimeSeriesHelper.cs:45]: Maybe it should be called MeterTimeSeriesHelper

File: domains/measurements/Measurements.API/API/Measurements/Helpers/TimeSeriesHelper.cs (hunk 7 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/Measurements/Helpers/TimeSeriesHelper.cs:57]: Maybe it should be called MeterTimeSeriesHelper

File: domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Dto/Responses/Enums/MeterType.cs (hunk 8 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Dto/Responses/Enums/MeterType.cs:8]: Done

File: domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Dto/Responses/MeteringPoint.cs (hunk 10 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Dto/Responses/MeteringPoint.cs:84]: AIB?

File: domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Dto/Responses/Address.cs (hunk 9 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Dto/Responses/Address.cs:3]: Dunno. This is taken from DataSync. I can make it a record if you want?

File: domains/measurements/Measurements.API/API/Shared/Exceptions/SanitizedException.cs (hunk 11 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/Shared/Exceptions/SanitizedException.cs:13]: So sanitize CPR numbers from exceptions

File: domains/measurements/Measurements.API/API/Program.cs (hunk 12 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/Program.cs:9]: To use correct appsettings.json depending on the environment

File: domains/measurements/Measurements.API/Tests/Extensions/NSubstituteExtensions.cs (hunk 13 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/Extensions/NSubstituteExtensions.cs:11]: I can't get Returns to work with asynchronous calls wihtout it

File: domains/measurements/Measurements.API/Tests/Fixtures/TestServerHelpers/ForwardLoggingProvider.cs (hunk 14 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/Fixtures/TestServerHelpers/ForwardLoggingProvider.cs:2]: Taken from the wallet, which got it from somewhere else. It basically says it's been copy pasted

File: domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Controllers/MeteringPointsController.cs (hunk 15 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Controllers/MeteringPointsController.cs:29]: Sure - you mean in the response summary there?

File: domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Controllers/MeteringPointsController.cs (hunk 16 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Controllers/MeteringPointsController.cs:46]: It maps the same way as DataSync.API. Don't think we are using all parameters

File: domains/measurements/Measurements.API/API/Startup.cs (hunk 17 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/Startup.cs:39]: No - transfer does not at least. I removed the 'options.JsonSerializerOptions.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;' line

File: domains/measurements/Measurements.API/API/Startup.cs (hunk 18 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/Startup.cs:69]: corrected

File: domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs (hunk 19 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs:2]: Taken from the wallet, which got it from somewhere else. It basically says it's been copy pasted

File: domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs (hunk 20 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs:44]: It's been copy pasted from the wallet which has it from somewhere else

File: domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs (hunk 21 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs:199]: Not sure - again copy paste

File: domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs (hunk 22 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/Fixtures/TestServerFixture.cs:55]: It makes a test server for gRPC and http

File: domains/measurements/Measurements.API/Tests/Fixtures/TestServerHelpers/TestServerContext.cs (hunk 24 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/Fixtures/TestServerHelpers/TestServerContext.cs:1]: Taken from the wallet, which got it from somewhere else. It basically says it's been copy pasted

File: domains/measurements/Measurements.API/Tests/Measurements/Helpers/TimeSeriesHelperTests.cs (hunk 25 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/Measurements/Helpers/TimeSeriesHelperTests.cs:21]: Cause then it is easy to extend in case of other cases

File: domains/measurements/Measurements.API/Tests/MeasurementsTestBase.cs (hunk 26 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/MeasurementsTestBase.cs:45]: Played with dispose - removed

File: domains/measurements/Measurements.API/Tests/MeteringPoints/Api/v2024_01_10/MeteringPointControllerTests.cs (hunk 27 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/MeteringPoints/Api/v2024_01_10/MeteringPointControllerTests.cs:22]: No. If we can I'm not sure how

File: domains/measurements/Measurements.API/Tests/MeteringPoints/Api/v2024_01_10/MeteringPointControllerTests.cs (hunk 28 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/MeteringPoints/Api/v2024_01_10/MeteringPointControllerTests.cs:192]: We could but this is only testing the trimming

File: domains/measurements/Measurements.API/Tests/Measurements/gRPC/V1/Services/MeasurementsServiceTests.GetMeasurements.verified.txt (hunk 29 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/Measurements/gRPC/V1/Services/MeasurementsServiceTests.GetMeasurements.verified.txt:7]: Not sure what you mean?

File: domains/measurements/Measurements.API/Tests/MeteringPoints/Api/v2024_01_10/MeteringPointControllerTests.GetMeteringPoints.verified.txt (hunk 30 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/MeteringPoints/Api/v2024_01_10/MeteringPointControllerTests.GetMeteringPoints.verified.txt:4]: It comes in the response but it is not mapped out to this verified file. The same goes SubMeterType. Do you know why?

File: domains/measurements/Measurements.API/Tests/Models/MeteringPointTests.cs (hunk 31 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/Models/MeteringPointTests.cs:24]: done

File: domains/measurements/Measurements.API/Tests/SwaggerTests.GetSwaggerDocs_v1.0.verified.txt (hunk 32 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/SwaggerTests.GetSwaggerDocs_v1.0.verified.txt:9]: This file is not used anymore after deletion of old endpoints

File: domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Dto/Responses/Address.cs (hunk 9 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Dto/Responses/Address.cs:3]: Made record

File: domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Dto/Responses/MeteringPoint.cs (hunk 10 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/API/MeteringPoints/Api/v2024_01_10/Dto/Responses/MeteringPoint.cs:84]: AssetType has been removed

File: domains/measurements/Measurements.API/Tests/MeteringPoints/Api/v2024_01_10/MeteringPointControllerTests.GetMeteringPoints.verified.txt (hunk 30 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/MeteringPoints/Api/v2024_01_10/MeteringPointControllerTests.GetMeteringPoints.verified.txt:4]: Made separate assert for this

File: domains/measurements/Measurements.API/Tests/Measurements/gRPC/V1/Services/MeasurementsServiceTests.GetMeasurements.verified.txt (hunk 29 above)
Comment (tnickelsen) [reply domains/measurements/Measurements.API/Tests/Measurements/gRPC/V1/Services/MeasurementsServiceTests.GetMeasurements.verified.txt:7]: Added specific assert for this

**Review by ckr123 — APPROVED** [review]
